### PR TITLE
Update the list of Perl editors

### PR DESCRIPTION
**Pull request by briandfoy**
For #83.

[0001-Update-the-list-of-Perl-editors-83.patch](https://github.com/perl-doc-cats/perlfaq/files/12862349/0001-Update-the-list-of-Perl-editors-83.patch)

### CODE REVIEW

**Comment by wchristian** [comment]
while komodo is not actively developed anymore, it can still be downloaded and still works, and provides some features others don't have, or provide only in a degraded fashion, such as fully featured realtime debugging

File: lib/perlfaq3.pod [hunk 1]
```diff
@@ -340,214 +340,78 @@ L<http://www.cpan.org/authors/id/T/TO/TOMC/scripts/toms.exrc.gz>
...
 
-If you're on Unix, you already have an IDE--Unix itself. The Unix
-philosophy is the philosophy of several small tools that each do one
-thing and do it well. It's like a carpenter's toolbox.
+You may want an editor that supports the language server protocol (LSP).
+There are a couple projects that supply the Perl intelligence:
 
-If you want an IDE, check the following (in alphabetical order, not
-order of preference):
+=ver 4
```
Comment (davorg) [inline lib/perlfaq3.pod:351]: `=over`

File: lib/perlfaq3.pod [hunk 2]
```diff
@@ -340,214 +340,78 @@ L<http://www.cpan.org/authors/id/T/TO/TOMC/scripts/toms.exrc.gz>
...
-L<http://space.mit.edu/~davis/jed/>
+=item L<Geany|https://www.geany.org>
 
-=back
-
-or a vi clone such as
+=item L<IntelliJ IDEA|https://www.jetbrains.com/idea/>
 
-=over 4
+USe the L<Perl plugin from the Jet Brains marketplace|https://plugins.jetbrains.com/plugin/7796>.
```
Comment (davorg) [inline lib/perlfaq3.pod:382]: "Use"

Comment (smonff) [reply lib/perlfaq3.pod:382]: "JetBrains", no space.

File: lib/perlfaq3.pod [hunk 3]
```diff
@@ -340,214 +340,78 @@ L<http://www.cpan.org/authors/id/T/TO/TOMC/scripts/toms.exrc.gz>
...
+No LSP support.
 
-=item XEmacs
+=item L<Eclipse|http://e-p-i-c.sf.net/>
 
-L<http://www.xemacs.org/Download/index.html>
+=item L<Emacs|https://www.gnu.org/software/emacs/download.html>
 
-=item Jed
+Use cperl-mode.
```
Comment (smonff) [inline lib/perlfaq3.pod:376]:
```suggestion
Use cperl-mode and L<lsp-mode|https://emacs-lsp.github.io/lsp-mode>.
```

**Review by smonff — CHANGES_REQUESTED** [review]

File: lib/perlfaq3.pod [hunk 4]
```diff
@@ -340,214 +340,78 @@ L<http://www.cpan.org/authors/id/T/TO/TOMC/scripts/toms.exrc.gz>
...
-=over 4
+=item L<Visual Studio Code|https://code.visualstudio.com/>
 
-=item BBEdit and TextWrangler
+Install the L<Perl plugin|https://marketplace.visualstudio.com/items?itemName=richterger.perl>
+from the Visual Studio Marketplace.
 
-are text editors for OS X that have a Perl sensitivity mode
-( L<http://www.barebones.com/> ).
+=item L<WebStorm|https://www.jetbrains.com/webstorm/>
```
Comment (hurricup) [inline lib/perlfaq3.pod:414]: Webstorm as well as other jetbrains IDEs  (RubyMine, PhpStorm, PyCharm, etc. see [supported products](https://plugins.jetbrains.com/plugin/7796-perl/versions/stable/412532)) supporting perl only with perl plugin mentioned above. Well, you may have a textmate highlighting without it, but that's it. Any code intelligence, profiling, coverage, debugging and so on provided by the plugin.

**Comment by wchristian** [comment]
to be clear: addressing mine and mst's comment is not optional

**Review by wchristian — CHANGES_REQUESTED** [review]
justify all deletions

**Comment by briandfoy** [comment]
Some people want to list Komodo, but it's abandoned and its own page says it has stability issues on macOS. There is no support for this editor and people just starting who need this advice shouldn't choose something that was abandoned two years ago. Other editors are likewise unsupported, abandoned, or too onerous to install for those who need this basic advice. 

It's not like people have a hard time finding a Windows IDE now that Visual Code supports Perl.

I've taken this patch as far as I'm going to take it. If the maintainers want to include different editors, they can contribute to the PR. I've added my parts, the community can add other parts. If someone wants a particular editor in the list, they can add their commit to the PR so people can see who suggested that editor.

**Comment by wchristian** [comment]
@briandfoy In the past you have claimed the nazi party of germany was left wing and justified perl-community-affecting action based on that. You were factually false then. Your action was wrong then. Your further response then was dehumanizing, dishonest, egotistical and refused actual dialogue.

Now much of this applies again.

I will be taking this patch forward so people will be guided by experience and knowledge.

**Comment by hurricup** [comment]
Why not just mention abandoned editors as abandoned editors user could use on their own risk?

**Comment by wchristian** [comment]
Yep, i will be explaining that. It is also not only abandoned, but open-sourced. And there are things visual code can't do right yet, which work just fine in Komodo.

**Comment by hurricup** [comment]
@wchristian try community Intellij IDEA with perl plugin. Both opensource and free. And smarter than Komodo.

**Comment by wchristian** [comment]
Last time i evaluated intellij, the perl plugin for it only provided crude debugging. But i will be evaluating things on the list to have up-to-date info.

**Pull request closed without merging**